stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/composer/MessageComposerController.kt

```diff
@@ -131,9 +155,9 @@ public class MessageComposerController(
     public fun setMessageInput(value: String) {
         this.input.value = value
 
-        setTyping(value.isNotEmpty())
-
-        validateInput()
+        sendTypingEventIfNecessary(isTyping = value.isNotEmpty())
```

Comment:
Honestly, I would rename all these to something like:

`handleTypingEvent()`, `handleMentionSuggestions()` etc. `IfNecessary` is lengthy, can be confusing and honestly is redundant, as we have conditions in the function that decide if it's necessary or not.

Also - this way we can handle multiple cases, and not just "sending events" and "not sending events".

Reply:
Agree 👍